helios-testing/src/main/java/com/spotify/helios/testing/HeliosSoloDeployment.java

```diff
@@ -410,10 +436,78 @@ public void close() {
   }
 
   /**
+   * Undeploy jobs left over by {@link TemporaryJobs}. TemporaryJobs should clean these up,
+   * but sometimes a few are left behind for whatever reason.
+   */
+  private void undeployLeftoverJobs() {
```

Comment:
(But I don't have a big problem with this, it's fine -- waiting for a job be undeployed when without calling undeploy (defensively) seems like a worse option :))